CppCoreGuidelines.md

```diff
@@ -11843,7 +11843,10 @@ Instead, prefer to put the common code in a common helper function -- and make i
 
         template<class T>           // good, deduces whether T is const or non-const
         static auto get_bar_impl(T& t) -> decltype(t.get_bar())
```

Comment:
I find it offputting (though it's not definitively incorrect) that the return type is a decltype of something that is not the return expression. Is the decltype necessary at all here now that the return value is explicit in the example?

Reply:
Depending on the call, the decltype will end up in const or non-const return value.

Reply:
I was skeptical about this but did some investigation and concluded you're 100% right. Thinking about it a bit more that's sort of the whole point of this example. Thanks for your patience.

Reply:
Ah so it didn't work because it was returning auto not auto& and therefore making a copy which would not be const.